core/platform/android/java/src/org/axmol/lib/AxmolGLSurfaceView.java

```diff
@@ -427,6 +401,38 @@ public void run() {
     // Methods
     // ===========================================================
 
+    public void handleOnResume() {
+        this.queueEvent(new Runnable() {
+            @Override
+            public void run() {
+                AxmolGLSurfaceView.this.mRenderer.handleOnResume();
+            }
+        });
+    }
+
+    public void handleOnPause() {
+        mNativePauseComplete = new CountDownLatch(1);
+
+        CountDownLatch complete = mNativePauseComplete;
+        this.queueEvent(new Runnable() {
+            @Override
+            public void run() {
+                AxmolGLSurfaceView.this.mRenderer.handleOnPause();
+                complete.countDown();
+            }
+        });
+    }
+
+    public void waitForPauseToComplete() {
+        while (mNativePauseComplete.getCount() > 0) {
+            try {
+                mNativePauseComplete.await();
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```

Comment:
Maybe just ignore the exception since you wait while latch count == 0

Reply:
Yes, nice catch! It was added by autocomplete and I missed it...